packages/react/popper/src/Popper.tsx

```diff
@@ -188,6 +189,14 @@ const PopperContent = React.forwardRef<PopperContentElement, PopperContentProps>
           : undefined,
         arrow ? floatingUIarrow({ element: arrow, padding: arrowPadding }) : undefined,
         avoidCollisions ? flip({ ...detectOverflowOptions }) : undefined,
+        size({
+          apply({ availableWidth, availableHeight, elements }) {
```

Comment:
`apply` also receives `middlewareData` which could be used to support the alternative approach I describe in the PR description.